From 78961d55c6481b50f6b160fdef38bd53f21e87ca Mon Sep 17 00:00:00 2001
From: Roland Rosenfeld <roland@debian.org>
Date: Tue, 26 Sep 2023 16:59:22 +0200
Subject: [PATCH 04/11] Work around issue with ghostscript 10.02 and
 -sPageList=1

GhostScript 10.02 fails in the call
gs -q -dSAFER -sDEVICE=eps2write -sPageList=1 -o - in.pdf
Using "-dFirstPage=1 -dLastPage=1" as a replacement for
"-sPageList=1" works around the issue for fig2dev.
---
 fig2dev/dev/readeps.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fig2dev/dev/readeps.c b/fig2dev/dev/readeps.c
index bddf3ca..d6b30a2 100644
--- a/fig2dev/dev/readeps.c
+++ b/fig2dev/dev/readeps.c
@@ -434,7 +434,7 @@ pdftops(struct xfig_stream *restrict pic_stream, FILE *out)
 	 * Convert a pdf to eps, using the first available of the following:
 	 *   pdftops -q -f 1 -l 1 -eps in.pdf -
 	 *   pdftocairo -q -f 1 -l 1 -eps in.pdf -
-	 *   gs -q -dSAFER -sDEVICE=eps2write -sPageList=1 -o - in.pdf
+	 *   gs -q -dSAFER -sDEVICE=eps2write -dFirstPage=1 -dLastPage=1 -o - in.pdf
 	 * Only pdftops uses the original /MediaBox as the BoundingBox in the
 	 * resulting pfd. The other two crop the BoundingBox to the smallest box
 	 * containing all the ink on the paper.
@@ -446,7 +446,7 @@ pdftops(struct xfig_stream *restrict pic_stream, FILE *out)
 #ifdef GSEXE
 	else if (has_gs())
 		cmd_fmt = GSEXE
-			" -q -dSAFER -sDEVICE=eps2write -sPageList=1 -o - '%s'";
+			" -q -dSAFER -sDEVICE=eps2write -dFirstPage=1 -dLastPage=1 -o - '%s'";
 #endif
 	else {
 		static bool	reported = false;
-- 
2.34.1

